.ci_support/build_all.py

```diff
@@ -207,9 +240,47 @@ def build_folders(recipes_dir, folders, arch, channel_urls):
         conda_build.api.build([recipe], config=get_config(arch, channel_urls))
 
 
+def build_folders_rattler_build(recipes_dir: str, platform, arch, channel_urls: list[str]):
+    config = get_config(arch, channel_urls)
+
+    # Remove the example recipes to ensure that they are not also build.
+    for example_recipe in EXAMPLE_RECIPE_FOLDERS:
+        rmtree(os.path.join(recipes_dir, example_recipe), ignore_errors=True)
+    
+    # Determine the locations for the variant config files.
+    specs = OrderedDict()
+    for f in config.exclusive_config_files:
+        specs[f] = conda_build.variants.parse_config_file(os.path.abspath(os.path.expanduser(os.path.expandvars(f))), config)
+
+    # Combine all the variant config files together
+    combined_spec = conda_build.variants.combine_specs(specs, log_output=config.verbose)
+    variant_config = yaml.dump(combined_spec)
+
+    # Define the arguments for rattler-build
+    args = [
+        "rattler-build",
+        "build", 
+        "--recipe-dir", recipes_dir,
+        "--target-platform", f"{platform}-{arch}",
+    ]
+    for channel_url in channel_urls:
+        # Local is automatically added by rattler-build so we just remove it.
+        if channel_url != "local":
+            args.extend(["-c", channel_url])
+    
+    # Construct a temporary file where we write the combined variant config. We can then pass that
+    # to rattler-build.
+    with tempfile.NamedTemporaryFile(delete=False) as fp:
```

Comment:
```suggestion
    with tempfile.NamedTemporaryFile(delete=False) as fp:
```

Maybe we should register this with `atexit` so it's removed once done.